src/common/FileSystem.cpp

```diff
@@ -1432,13 +1434,19 @@ void CopyFile(Str::StringRef path, const File& dest, std::error_code& err)
 		char buffer[65536];
 		while (true) {
 			offset_t read = zipFile.ReadFile(buffer, sizeof(buffer), err);
-			if (err)
+			if (err) {
+				std::error_code ignored;
+				zipFile.CloseFile(ignored);
```

Comment:
So there's this wacky multi-mode error handling thing in these functions... 
If `&err` equals a magic value, then errors are reported by throwing exceptions, instead of by setting the error code. So CloseFile will not get called in exception mode.

I'd still accept the PR which is an improvement over the current state. Maybe add a TODO about the fact that it doesn't work when using exceptions.

Probably we should get rid of this double paradigm error stuff because it is too hard to write correctly.

Reply:
Done.